File: MainActivity.java
package com.example.sanjay.tipcalculatorjava;

import androidx.appcompat.app.AppCompatActivity;

import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.EditText;
import android.widget.RadioButton;
import android.widget.RadioGroup;
import android.widget.Switch;
import android.widget.TextView;

import java.text.NumberFormat;

public class MainActivity extends AppCompatActivity
{
    double bill;
    protected void onCreate(Bundle savedInstanceState)
    {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
    }
    public void onclickrbtn(View view)
    {
        EditText editText = findViewById(R.id.costofservice);
        double amount = Double.parseDouble(editText.getText().toString());
        RadioButton amazing = findViewById(R.id.twentypercent);
        RadioButton good = findViewById(R.id.eighteenpercent);
        RadioButton okay = findViewById(R.id.fiteenpercent);
        TextView tip = findViewById(R.id.tipamt);
        boolean checked = ((RadioButton) view).isChecked();
        Button button = findViewById(R.id.btn);
        button.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                switch (view.getId()){
                    case R.id.twentypercent:
                        if (checked)
                            bill = (amount + (amount*.20));
                            tip.setText("$"+bill);
                        break;
                    case R.id.eighteenpercent:
                        if (checked)
                            bill = (amount + (amount*.18));
                        tip.setText("$"+" "+bill);
                        break;
                    case R.id.fiteenpercent:
                        if (checked)
                            bill = (amount + (amount*.15));
                        tip.setText("$"+bill);
                        break;
                }
            }
        });
    }

}

File: src/main/java/com/example/sanjay/DiceRoller/MainActivity.kt
package com.example.sanjay.DiceRoller

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.widget.Button
import android.widget.TextView
import android.widget.Toast

class MainActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        val rollbutton: Button = findViewById(R.id.button)
        rollbutton.setOnClickListener {
            Toast.makeText(this,"Dice Rolled",Toast.LENGTH_SHORT).show()
            Thread.sleep(1000)
            RollDice()
        }


    }

    private fun RollDice() {
         val dice = Dice(6)
        val diceroll=dice.Roll()
        val txtview: TextView = findViewById(R.id.textView)
        txtview.text = diceroll.toString()




    }
}
class Dice(val numsides:Int){
    fun Roll() :Int{
        return (1..numsides).random()
    }
}